src/psbt.cpp

```diff
@@ -240,37 +240,51 @@ bool SignPSBTInput(const SigningProvider& provider, PartiallySignedTransaction&
     SignatureData sigdata;
     input.FillSignatureData(sigdata);
 
-    // Get UTXO
+    // Get UTXOs
     bool require_witness_sig = false;
-    CTxOut utxo;
-
-    if (input.non_witness_utxo) {
-        // If we're taking our information from a non-witness UTXO, verify that it matches the prevout.
-        COutPoint prevout = tx.vin[index].prevout;
-        if (prevout.n >= input.non_witness_utxo->vout.size()) {
-            return false;
+    bool have_all_spent_outputs = true;
+    std::vector<CTxOut> utxos(tx.vin.size());
+    for (size_t idx = 0; idx < tx.vin.size(); ++idx) {
```

Comment:
In 1d3cc29fb86b614b76b4d97a5adab7f903fc3020 "Construct and use PrecomputedTransactionData in PSBT signing"

We should avoid iterating all of the PSBT inputs again when we are signing each input. Instead we should compute the `PrecomputedTransactionData` in the caller and pass that into `SignPSBTInput`.

This can be done in `CWallet::FillPSBT` where we are iterating the inputs anyways to get their previous txs from the wallet. We could be constructing the `PrecomputedTransactionData` at that time and then passing it into `FIllPSBT` which then passes it to `SignPSBTInput`.

Reply:
Done.